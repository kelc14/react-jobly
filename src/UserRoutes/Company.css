.Company-container {
  width: 70vw;
  background-color: white;
  border: 1px solid black;
  border-radius: 10px;
  margin: 0 auto;
  padding: 20px;
  overflow: hidden;
  text-align: left;
  margin-bottom: 25px;
}

.Company-company {
  text-align: left;
  font-weight: bold;
  font-size: 1rem;
}

.Company-about {
  text-align: left;
  font-size: 0.8rem;
}

.Company-logo {
  float: right;
}
