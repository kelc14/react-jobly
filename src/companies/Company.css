.Company-container {
  width: 70vw;
  background-color: white;
  border: 1px solid black;
  border-radius: 10px;
  margin: 0 auto;
  padding: 20px;
  overflow: hidden;
  text-align: left;
  margin-bottom: 25px;
}

.Company-container:hover {
  background-color: goldenrod;
}

.Company-link {
  text-decoration: none;
  color: black;
}

.Company-company {
  text-align: left;
  font-weight: bold;
  font-size: 1rem;
}

.Company-about {
  text-align: left;
  font-size: 0.8rem;
}

.Company-logo {
  float: right;
}
