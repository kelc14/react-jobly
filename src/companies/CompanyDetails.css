.CompanyDetails-company {
  margin: 40px auto;

  font-family: "Ubuntu", sans-serif;
  font-size: 1.5rem;
  font-weight: bold;
  color: goldenrod;
  letter-spacing: 5px;
  text-shadow: 2px 7px 5px rgba(0, 0, 0, 0.3),
    0px -4px 10px rgba(255, 255, 255, 0.3);
}

.CompanyDetails-about {
  margin: 40px auto;
  margin-bottom: 10px;

  font-family: "Ubuntu", sans-serif;
  font-size: 1rem;
  font-weight: bold;
  color: goldenrod;
}

.CompanyDetails-error-container {
  width: 70vw;
  background-color: white;
  border: 1px solid black;
  border-radius: 10px;
  margin: 0 auto;
  margin-top: 25px;
  padding: 20px;
  overflow: hidden;
  text-align: center;
  position: relative;
  margin-bottom: 25px;
}
.CompanyDetails-error {
  color: black;
  font-family: "Ubuntu", sans-serif;
  font-size: 1rem;
  margin-bottom: 25px;
}

.CompanyDetails-error-container a {
  background-color: goldenrod;
  padding: 10px;
  margin: 20px;
  border-radius: 8px;
  text-decoration: none;
  color: white;
  font-weight: 500;
  /* text-shadow: 2px 7px 5px rgba(0, 0, 0, 0.3),
  0px -4px 10px rgba(255, 255, 255, 0.3); */
}

.CompanyDetails-error-container a:hover {
  background-color: white;
  border: 1px solid goldenrod;
  color: goldenrod;
}
