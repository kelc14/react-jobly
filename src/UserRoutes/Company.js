import React from "react";
import "./Company.css";

const Company = () => {
  return (
    <div className="Company-container">
      <p className="Company-company"> Company Name Goes Here </p>
      <p className="Company-about"> Description: 1,0000000 </p>
      <p className="Company-about"> Number of Employees: 00000 </p>
      <img src={`img logo here`} alt="logo" className="Company-logo" />
    </div>
  );
};

export default Company;
