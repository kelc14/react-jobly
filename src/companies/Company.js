import React from "react";
import "./Company.css";
import { Link } from "react-router-dom";

const Company = ({ data }) => {
  const { name, handle, description, numEmployees, logoUrl } = data;
  return (
    <Link to={`./${handle}`} className="Company-link">
      <div className="Company-container">
        <p className="Company-company"> {name} </p>
        <p className="Company-about"> Description: {description} </p>
        <p className="Company-about"> Number of Employees: {numEmployees} </p>
        {logoUrl ? (
          <img src={`${logoUrl}`} alt="logo" className="Company-logo" />
        ) : (
          <></>
        )}
      </div>{" "}
    </Link>
  );
};

export default Company;
