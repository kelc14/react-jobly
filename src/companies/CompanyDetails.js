import React, { useState, useEffect } from "react";
import "./CompanyDetails.css";
import { useParams, Link } from "react-router-dom";
import useErrors from "../hooks/useErrors";

import JoblyApi from "../api/api";
import Job from "../jobs/Job";

/** Company details page:
 *
 * displays information about the company - name, description - and list of all jobs (Job components)
 */

const CompanyDetails = ({ addNewJob }) => {
  const { handle } = useParams();
  const [company, setCompany] = useState({});
  const [jobs, setJobs] = useState([]);

  const [err, documentErrors, showError] = useErrors();

  useEffect(() => {
    const getCompanyByHandle = async () => {
      try {
        let res = await JoblyApi.getCompany(handle);
        setCompany(() => res);
        setJobs(() => res.jobs);
      } catch (e) {
        documentErrors(e);
      }
    };
    getCompanyByHandle();
  }, [handle]);

  return (
    <div className="CompanyDetails-container">
      <p className="CompanyDetails-company"> {company.name} </p>
      <p className="CompanyDetails-about">{company.description} </p>
      {jobs.length === 0 ? (
        <div className="CompanyDetails-error-container">
          <p className="CompanyDetails-error">
            Sorry, no jobs to display. Check back later.
          </p>
          {err && showError()}

          <Link to="/companies">View All Companies</Link>
        </div>
      ) : (
        jobs.map((job) => (
          <Job
            jobData={job}
            companiesData={company}
            addNewJob={addNewJob}
            key={job.id}
          />
        ))
      )}
    </div>
  );
};

export default CompanyDetails;
