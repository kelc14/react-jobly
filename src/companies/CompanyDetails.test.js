import UserContext from "../hooks/UserContext";
import { render, act } from "@testing-library/react";
import CompanyDetails from "./CompanyDetails";
import JoblyApi from "../api/api";
import Company from "./Company";
import { MemoryRouter } from "react-router-dom";

let testCompany = {
  company: {
    handle: "test-company",
    name: "Test Company",
    description: "This is a test company for mocking our API calls.",
    numEmployees: 1000,
    logoUrl: null,
    jobs: [
      {
        id: 66,
        title: "Test Job 1",
        salary: 100000,
        equity: 0,
        company_handle: "test-company",
      },
    ],
  },
};
/**
 * SMOKE TEST
 */
test("CompanyDetails renders without crashing", () => {
  // Mock the getCompany function
  act(() => {
    const getCompanyMock = jest.fn().mockResolvedValue(testCompany);

    jest.spyOn(JoblyApi, "getCompany").mockImplementation(getCompanyMock);
  });

  render(
    <MemoryRouter initialEntries={["/companies/test-company"]}>
      {" "}
      <CompanyDetails />
    </MemoryRouter>
  );
});

/**
 * SNAPSHOT TEST
 */
test("CompanyDetails matches snapshot", () => {
  // Mock the getCompany function
  act(() => {
    const getCompanyMock = jest.fn().mockResolvedValue(testCompany);

    jest.spyOn(JoblyApi, "getCompany").mockImplementation(getCompanyMock);
  });

  const { asFragment } = render(
    <MemoryRouter initialEntries={["/companies/test-company"]}>
      {" "}
      <CompanyDetails />
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});
