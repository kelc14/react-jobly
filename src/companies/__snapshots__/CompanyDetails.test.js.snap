// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CompanyDetails matches snapshot 1`] = `
<DocumentFragment>
   
  <div
    class="CompanyDetails-container"
  >
    <p
      class="CompanyDetails-company"
    >
        
    </p>
    <p
      class="CompanyDetails-about"
    >
       
    </p>
    <div
      class="CompanyDetails-error-container"
    >
      <p
        class="CompanyDetails-error"
      >
        Sorry, no jobs to display. Check back later.
      </p>
      <a
        href="/companies"
      >
        View All Companies
      </a>
    </div>
  </div>
</DocumentFragment>
`;
